near2eth/near2eth-block-relay/index.js

```diff
@@ -186,10 +188,53 @@ class Near2EthRelay {
     const maxDelay = Number(near2ethRelayMaxDelay)
     const errorDelay = Number(near2ethRelayErrorDelay)
 
+    const selectDuration = web3.utils.toBN(Number(near2ethRelaySelectDuration) * 1000_000_000)
+
+    ethGasMultiplier = Number(ethGasMultiplier)
+    ethUseEip1559 = ethUseEip1559 === 'true'
+
     const httpPrometheus = new HttpPrometheus(this.metricsPort, 'near_bridge_near2eth_')
     const clientHeightGauge = httpPrometheus.gauge('client_height', 'amount of block client processed')
     const chainHeightGauge = httpPrometheus.gauge('chain_height', 'current chain height')
 
+    const nextBlockSelection = {
+      startedAt: 0,
+      borshBlock: null,
+      height: 0,
+      set: function ({ borshBlock, lightClientBlock }) {
+        if (this.isEmpty()) {
+          this.startedAt = lightClientBlock.inner_lite.timestamp
+        }
+        this.borshBlock = borshBlock
+        this.height = lightClientBlock.inner_lite.height
+        console.log(`The new optimal block is found. Height: ${this.height}. Size: ${this.borshBlock.length} bytes`)
+      },
+      clean: function () {
+        this.startedAt = 0
+        this.borshBlock = null
+        this.height = 0
+      },
+      isEmpty: function () {
+        return !this.borshBlock
+      },
+      isSuitable: function ({ borshBlock, lightClientBlock }) {
+        return !this.isEmpty() &&
+               this.height !== lightClientBlock.inner_lite.height &&
+               this.borshBlock.length >= borshBlock.length
```

Comment:
I'd say the `lightClientBlock` is suitable if
```js
this.isEmpty() 
|| (!this.isEmpty()
   && this.height !== lightClientBlock.inner_lite.height
   && this.borshBlock.length >= borshBlock.length)
```

as the block is suitable if there's no more block, isn't it? If we use it, then we can simplify branches with block-selection (when `isEmpty()` and `isSuitable()` are called.

Reply:
The logic for selecting the suitable block was moved outside the `nextBlockSelection` object for a reason. This is done to preserve the "straightforward" code style of the main loop. This non-obvious decision was commented on [line 271](https://github.com/aurora-is-near/rainbow-bridge/pull/735/files#diff-4aa209577ae540426c39ef90f8b1e1b64b5bd96613ce4f144479bab1b2843a40R271).  I hope there are plans for refactoring :) But I agree. Will be fixed.